app/hotels/src/allHotels/HotelsPaginationContainer.js

```diff
@@ -35,16 +38,13 @@ export function HotelsPaginationContainer(props: Props) {
   }, []);
 
   React.useEffect(() => {
-    const priceMax = props.data?.allAvailableBookingComHotels?.stats?.maxPrice;
-    const priceMin = props.data?.allAvailableBookingComHotels?.stats?.minPrice;
-
     if (priceMax != null && priceMin != null) {
       setCurrentSearchStats({
-        priceMax,
-        priceMin,
+        priceMax: new Decimal(priceMax),
+        priceMin: new Decimal(priceMin),
       });
     }
-  }, [props.data, setCurrentSearchStats]);
+  }, [priceMax, priceMin, setCurrentSearchStats]);
```

Comment:
That looks a lot clearer, I like that!

Reply:
It is also a performance improvement, since it is no longer diffing the entire `props.data` 😄